### PR TITLE
Refactor Prometheus exporter

### PR DESCRIPTION
**Pull request by brfrn169**
The changes are as follows:
- Rename the config name `scalar.db.server.prometheus_http_endpoint_port` to `scalar.db.server.prometheus_exporter_port`
- Change the behavior to disable the Prometheus exporter when a negative number port is given
- Change the Prometheus HTTP endpoint path from `/metrics` to `/stats/prometheus`

Please take a look!

### CODE REVIEW

File: server/src/main/java/com/scalar/db/server/service/ServerModule.java [hunk 1]
```diff
@@ -77,23 +77,23 @@ private void startJmxReporter(MetricRegistry metricRegistry) {
...
     }
 
     CollectorRegistry.defaultRegistry.register(new DropwizardExports(metricRegistry));
 
-    Server server = new Server(prometheusHttpEndpointPort);
+    Server server = new Server(prometheusExporterPort);
     ServletContextHandler context = new ServletContextHandler();
     context.setContextPath("/");
     server.setHandler(context);
     context.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
```
Comment (feeblefakie) [inline server/src/main/java/com/scalar/db/server/service/ServerModule.java:92]: I made Scalar DL's endpoint to match with the one of Envoy.
https://www.envoyproxy.io/docs/envoy/latest/operations/admin#get--stats-prometheus

```suggestion
    context.addServlet(new ServletHolder(new MetricsServlet()), "/stats/prometheus");
```

**Review by feeblefakie — COMMENTED** [review]
Overall looking good. Left one suggestion.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by brfrn169** [comment]
It looks like the multi-storage integration test is failing for some reason. I will check it.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**